File: codes/cpp/1332.cpp
#include "global.hpp"

class Solution
{
    bool ifPalindromic(string &s, int start, int end)
    {
        for (int i = start, j = end; i < j; i++, j--)
        {
            if (s[i] != s[j])
                return false;
        }
        return true;
    }

public:
    int removePalindromeSub(string s)
    {
        if (ifPalindromic(s, 0, s.length() - 1))
        {
            return 1;
        }
        return 2;
    }
};

int main()
{
    Solution sol;
    cout << sol.removePalindromeSub("ababa") << endl;
    cout << sol.removePalindromeSub("abb") << endl;
    cout << sol.removePalindromeSub("baabb") << endl;
    cout << sol.removePalindromeSub("bbaabaaa") << endl;
}

File: codes/rust/1332.rs
impl Solution {
    pub fn remove_palindrome_sub(s: String) -> i32 {
        let s: Vec<char> = s.chars().collect();
        let (mut l, mut r) = (0, s.len() - 1);
        while l < r {
            if s[l] != s[r] {
                return 2;
            }
            l += 1;
            r -= 1;
        }
        1
    }
}
